src/flow/src/engine.rs

```diff
@@ -49,6 +49,8 @@ pub trait FlowEngine {
     async fn flush_flow(&self, flow_id: FlowId) -> Result<usize, Error>;
     /// Check if the flow exists
     async fn flow_exist(&self, flow_id: FlowId) -> Result<bool, Error>;
+    /// List all flows
+    async fn list_flows(&self) -> Result<Vec<FlowId>, Error>;
```

Comment:
From existing usages, I'd suggest to return `IntoIter<FlowId>` instead of a `Vec`